manager.go

```diff
@@ -0,0 +1,248 @@
+type Identity interface {
+	// Allow reports whether the user should be allowed to access
+	// any of the users or groups in the given ACL slice.
+	Allow(ctx context.Context, acl []string) (bool, error)
+}
+
+// AdminACL holds the name of the administrator ACL.
+const AdminACL = "admin"
+
+// CodeACLNotFound holds the error code returned from
+// the HTTP endpoints when an ACL name has not been
+// created.
+const CodeACLNotFound = "ACL not found"
+
+// Manager implements an ACL manager.
+type Manager struct {
+	p      Params
+	router *httprouter.Router
+}
+
+var errAuthenticationFailed = errgo.Newf("authentication failed")
+
+var reqServer = &httprequest.Server{
+	ErrorWriter: func(ctx context.Context, w http.ResponseWriter, err error) {
+		if errgo.Cause(err) == errAuthenticationFailed {
+			// The Authenticate method has already written its response.
+			return
+		}
+		log.Printf("writing error %#v", err)
+		status, body := errorMapper(ctx, err)
+		httprequest.WriteJSON(w, status, body)
+	},
+}
+
+func errorMapper(ctx context.Context, err error) (int, interface{}) {
+	switch errgo.Cause(err) {
+	case ErrACLNotFound:
+		return http.StatusNotFound, &httprequest.RemoteError{
+			Message: err.Error(),
+			Code:    CodeACLNotFound,
+		}
+	case ErrBadUsername:
+		err = httprequest.Errorf(httprequest.CodeBadRequest, "%v", err)
+	}
+	return httprequest.DefaultErrorMapper(ctx, err)
+}
+
+// NewManager returns a new Manager instance that manages a
+// set of ACLs. It ensures there is at least one ACL
+// created, named "admin", which is given p.InitialAdminUsers
+// when it is first created.
+func NewManager(ctx context.Context, p Params) (*Manager, error) {
+	if err := p.Store.CreateACL(ctx, AdminACL, p.InitialAdminUsers...); err != nil {
+		return nil, errgo.Notef(err, "cannot create initial admin ACL")
+	}
+	m := &Manager{
+		p:      p,
+		router: httprouter.New(),
+	}
+	httprequest.AddHandlers(m.router, reqServer.Handlers(m.newHandler))
+	return m, nil
+}
+
+// ServeHTTP implements http.Handler by serving an ACL administration
+// interface that allows clients to manipulate the ACLs. The set of
+// ACLs that can be manipulated can be changed with the Manager.NewACL
+// method.
+//
+// All the endpoints are situated underneath the RootPath prefix
+// passed to NewManager.
+func (m *Manager) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	path := req.URL.Path
+	if m.p.RootPath != "" {
+		path = strings.TrimPrefix(path, m.p.RootPath)
+		if len(path) == len(req.URL.Path) || path == "" || path[0] != '/' {
+			httprequest.WriteJSON(w, http.StatusNotFound, &httprequest.RemoteError{
+				Message: "URL path not found",
+				Code:    httprequest.CodeNotFound,
+			})
+			return
+		}
+	}
+	req.URL.Path = path
+	m.router.ServeHTTP(w, req)
+}
+
+// ACL returns the members of the given ACL.
+func (m *Manager) ACL(ctx context.Context, name string) ([]string, error) {
+	// TODO implement a cache to avoid hitting the underlying
```

Comment:
better to make a caching ACLStore wrapper and not worry about it here.

Reply:
I'm not sure. That depends a bit on whether the caching can be a closed abstraction or whether it might require HTTP endpoints of its own (for example to find out resident cache size or force a cache flush). I'll leave the TODO for now if that's ok, and we can chat about it when the time comes.